dapp/client/App.elm

```diff
@@ -67,19 +73,33 @@ update : Message -> Model -> ( Model, Cmd Message )
 update message model =
     case message of
         AddressChanged newAddress ->
-            ( { model | address = Just newAddress }, Cmd.none )
+            ( { initialModel | address = Just newAddress }, Cmd.none )
+
+        WalletChanged newWallet ->
+            ( { model | wallet = Just newWallet, eth = Nothing }, Cmd.none )
```

Comment:
As this continues to grow, we'll likely need to reset more and more.  I wonder if it would be easier when `WalletChanged` occurs to instead reset to `initialModel` and then fire off a sequence of `AddressChanged` and then `WalletChanged`?  Or perhaps do `{ initialModel | address = model.address, wallet = Just newWallet }` (I'm not actually sure what the syntax looks like for the latter).

My concern is that a wallet change means "reset everything except for the address and re-fetch all wallet-specific data", and I'm worried that this line will grow quite large and become difficult to maintain.